mlos_bench/mlos_bench/services/config_persistence.py

```diff
@@ -176,22 +180,28 @@ def load_config(
         config : Union[dict, List[dict]]
             Free-format dictionary that contains the configuration.
         """
-        json_file_name = self.resolve_path(json_file_name)
-        _LOG.info("Load config: %s", json_file_name)
-        with open(json_file_name, mode="r", encoding="utf-8") as fh_json:
-            config = json5.load(fh_json)
+        if any(c in json for c in ("{", "[")):
+            # If the path contains braces, it is likely already a json string,
+            # so just parse it.
+            _LOG.info("Load config from json string: %s", json)
+            config: Any = json5.loads(json)
+        else:
+            json = self.resolve_path(json)
+            _LOG.info("Load config file: %s", json)
+            with open(json, mode="r", encoding="utf-8") as fh_json:
+                config = json5.load(fh_json)
```

Comment:
This is the main change to allow testable docstrings with inline JSON configs.
Basically we just look at the provided string and see if it has a few non-path but JSON characters in it and then attempt to parse the string directly.